Add Dependabot documentation (#7141)
woocommerce/woocommerce-gutenberg-products-block: docs/internal-developers/automatations/README.md

> # Block Client APIs

Reviewer: I assume this was a copy-paste mistake? 🤔

> This folder contains documentation for all automations that are configurated in this repository.

Reviewer: ```suggestion
This directory contains documentation for all the automations configured in this repository.
```

| Document                                                                                                     | Description                                                                                                 |
| ------------------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------- |
| [Dependabot](dependabot.md)                                                                                  | This doc explains how Dependabot works.															         |

> 

Reviewer: Does this need to be this long? Can't we make the columns way shorter?

<!-- FEEDBACK -->

---

[We're hiring!](https://woocommerce.com/careers/) Come work with us!

🐞 Found a mistake, or have a suggestion? [Leave feedback about this document here.](https://github.com/woocommerce/woocommerce-blocks/issues/new?assignees=&labels=type%3A+documentation&template=--doc-feedback.md&title=Feedback%20on%20./docs/internal-developers/block-client-apis/README.md)

<!-- /FEEDBACK -->

> 

Reviewer: Directory name is misspelled: `automatations` → `automations`
